File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        int n = nums.size();

        int sequentialSum = nums[0];

        //finding longest Prefix Sequential Sum
        for(int i = 1; i < n; i++){
            if(nums[i] == nums[i - 1] + 1)
                sequentialSum += nums[i];
            else
                break;
        }

        vector<bool> hashTable(1276, false);

        //pulating hashTable
        for(int num : nums)
            hashTable[num] = true;

        //finding the missing number using ht
        while(hashTable[sequentialSum])
            sequentialSum++;

        return sequentialSum;
    }
};

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution:
    def missingInteger(self, nums):
        sequentialSum = nums[0]

        ##finding longest Prefix Sequential Sum
        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1] + 1:
                sequentialSum += nums[i]
            else:
                break

        hashTable = [False] * 1276

        #pulating hashTable
        for num in nums:
            hashTable[num] = True

        ##finding the missing number using ht
        while hashTable[sequentialSum]:
            sequentialSum += 1

        return sequentialSum